Formula/brew-up.rb

```diff
@@ -0,0 +1,18 @@
+class BrewUp < Formula
+  desc "Interactive upgrade outdated casks and outdated"
+  homepage "https://github.com/mkanako/brew-up"
+  url "https://github.com/mkanako/brew-up/archive/v1.0.0.tar.gz"
+  sha256 "b3f313304d6e45007953824fff167f5cc571c2d4f641ea3e6f15b93b60feadf7"
+  license "MIT"
+
+  bottle :unneeded
+
+  def install
+    bin.install "brew-up.sh"
+    bin.install_symlink "brew-up.sh" => "brew-up"
+  end
+
+  test do
+    assert_match "usage: brew-up [-s]", shell_output("#{bin}/brew-up -h")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.